Sources/MapboxNavigation/NavigationMapView+VanishingRouteLine.swift

```diff
@@ -175,15 +175,19 @@ extension NavigationMapView {
     /**
      Updates the route style layer and its casing style layer to gradually disappear as the user location puck travels along the displayed route.
      
+     - parameter coordinate: Current position of the user location.
      - parameter routeProgress: Current route progress.
      */
-    public func updateRoute(_ routeProgress: RouteProgress) {
-        let mainRouteLayerIdentifier = routeProgress.route.identifier(.route(isMainRoute: true))
-        let mainRouteCasingLayerIdentifier = routeProgress.route.identifier(.routeCasing(isMainRoute: true))
+    public func updateVanishingRouteLine(coordinate: CLLocationCoordinate2D?) {
```

Comment:
If `updateRoute(_:)` is too ambiguous, consider `travelAlongRouteLine(to:)`, to evoke the same terminology as `updateFractionTraveled(_:)` above.

The term “vanishing route line” is sort of an internal shorthand for the visual effect we have by default, but I think it could be confusing to bake it into a symbol name in the public API. For example, a developer may style the route line so that it changes colors but doesn’t vanish behind the puck. For that reason, the option that enables or disables this feature is called `routeLineTracksTraversal` in an effort to be more descriptive of behavior and less prescriptive about look and feel.

If `routeLineTracksTraversal` is set to `false`, does client code still need to update the route line based on progress, or we sure this functionality is only needed for when the vanishing route line feature is enabled?

Reply:
No, this method is only called when `routeLineTracksTraversal ` is set to `true`. Because it only works is to providing the vanishing effect for the route line. Thus if the customer want to change colors or the route line, this method would not be used.